src/plugins/discover/public/application/view_components/context/index.tsx

```diff
@@ -18,12 +19,12 @@ const SearchContext = React.createContext<SearchContextValue>({} as SearchContex
 // eslint-disable-next-line import/no-default-export
 export default function DiscoverContext({ children }: React.PropsWithChildren<ViewProps>) {
   const services = getServices();
-  const searchParams = useSearch(services);
-
   const {
-    services: { osdUrlStateStorage },
+    services: { osdUrlStateStorage, store },
   } = useOpenSearchDashboards<DataExplorerServices>();
 
+  const searchParams = useSearch(services, store);
```

Comment:
This change wont be necessary with my change here: https://github.com/opensearch-project/OpenSearch-Dashboards/pull/5168/files#diff-663ffea1a92b4c32f69ad6a65f15d9b07ae759bc026568db627d950842fcad07